src/components/application_manager/src/message_helper/message_helper.cc

```diff
@@ -1025,46 +1025,80 @@ smart_objects::SmartObjectSPtr MessageHelper::CreateSetAppIcon(
 }
 
 smart_objects::SmartObjectSPtr
-MessageHelper::CreateOnButtonSubscriptionNotification(
-    uint32_t app_id,
-    hmi_apis::Common_ButtonName::eType button,
-    bool is_subscribed,
-    ApplicationManager& app_mngr) {
+MessageHelper::CreateButtonSubscriptionHandlingRequestToHmi(
+    const uint32_t app_id,
+    const hmi_apis::Common_ButtonName::eType button,
+    const hmi_apis::FunctionID::eType function,
+    application_manager::ApplicationManager& app_mngr) {
   using namespace smart_objects;
   using namespace hmi_apis;
   LOG4CXX_AUTO_TRACE(logger_);
 
-  SmartObjectSPtr notification_ptr =
-      std::make_shared<SmartObject>(SmartType_Map);
-  if (!notification_ptr) {
-    LOG4CXX_ERROR(logger_, "Memory allocation failed.");
-    return SmartObjectSPtr(nullptr);
-  }
-  SmartObject& notification = *notification_ptr;
+  SmartObjectSPtr request_ptr = CreateMessageForHMI(
+      hmi_apis::messageType::request, app_mngr.GetNextHMICorrelationID());
 
+  SmartObject& request = *request_ptr;
   SmartObject msg_params = SmartObject(SmartType_Map);
   msg_params[strings::app_id] = app_id;
-  msg_params[strings::name] = button;
-  msg_params[strings::is_suscribed] = is_subscribed;
+  msg_params[strings::button_name] = button;
+  request[strings::params][strings::function_id] = function;
+  request[strings::msg_params] = msg_params;
+  return request_ptr;
+}
 
-  notification[strings::params][strings::message_type] =
-      static_cast<int32_t>(application_manager::MessageType::kNotification);
-  notification[strings::params][strings::protocol_version] =
-      commands::CommandImpl::protocol_version_;
-  notification[strings::params][strings::protocol_type] =
-      commands::CommandImpl::hmi_protocol_type_;
-  notification[strings::params][strings::function_id] =
-      hmi_apis::FunctionID::Buttons_OnButtonSubscription;
-  notification[strings::msg_params] = msg_params;
+smart_objects::SmartObjectSPtr MessageHelper::CreateButtonNotificationToMobile(
+    ApplicationManager& app_mngr,
+    ApplicationSharedPtr app,
+    const smart_objects::SmartObject& source_message) {
+  LOG4CXX_AUTO_TRACE(logger_);
 
-  return notification_ptr;
+  if (!app) {
+    LOG4CXX_ERROR(logger_, "application NULL pointer");
+    return nullptr;
```

Comment:
@mked-luxoft Are you sure that `nullptr` is good here? May return SmartObject ptr with SmartType_Null?

Reply:
@AByzhynar fixed in 2b001c6014447a350d41a2bcd1e834fd2c83057a